compose/Makefile

```diff
@@ -111,12 +111,29 @@ restart-am-services:  ## Restart Archivematica services: MCPServer, MCPClient, D
 	docker-compose restart archivematica-dashboard
 	docker-compose restart archivematica-storage-service
 
-compile-requirements:  ## Run pip-compile
+compile-requirements-am:
 	docker-compose run --workdir /src/MCPServer/requirements --volume "tmpfs:/pip-cache:rw" -e XDG_CACHE_HOME=/pip-cache --rm  --no-deps --user=root --entrypoint make archivematica-mcp-server all
 	docker-compose run --workdir /src/MCPClient/requirements --volume "tmpfs:/pip-cache:rw" -e XDG_CACHE_HOME=/pip-cache --rm --no-deps --user=root --entrypoint make archivematica-mcp-client all
 	docker-compose run --workdir /src/dashboard/src/requirements --volume "tmpfs:/pip-cache:rw" -e XDG_CACHE_HOME=/pip-cache --rm --no-deps --user=root --entrypoint make archivematica-dashboard all
 	docker-compose run --workdir /src/requirements --volume "tmpfs:/pip-cache:rw" -e XDG_CACHE_HOME=/pip-cache --rm  --no-deps --user=root --entrypoint make archivematica-storage-service all
 
+compile-requirements-ss:
+	docker-compose run --workdir /src/requirements --volume "tmpfs:/pip-cache:rw" -e XDG_CACHE_HOME=/pip-cache --rm  --no-deps --user=root --entrypoint make archivematica-storage-service all
+
+CALLER_UID=${USER}
+ifneq ($(strip $(SUDO_UID)),)
+	CALLER_UID=$(SUDO_UID)
+endif
+
+reset-permissions-to-user:
+	@echo "Running recipe 'reset-permissions-to-user' and setting ownership to ${CALLER_UID} -- if eq 1000 this should be your USER"
+	find ../src/ -user root -path "*/requirements/*" -exec sudo chmod 644 '{}' +
```

Comment:
Would this require a `sudo` password when this make rule is executed?

Reply:
@replaceafill that's  right,  yep. You can `sudo` the recipe, or supply it during its execution.

Reply:
It'd be nice if we never changed the permissions in the first place. In compile-requirements-*, we can use `docker-compose run --user=${CALLER_UID}` instead of `--user=root`. That way we wouldn't need to reset permissions? Well I guess users already running am where we messed up with their permissions will have to reset them one last time.

Reply:
I have replaced this recipe with one that does the work in the container instead which immediately I think is better than asking the Makeflle to do something at the system level. Regarding the compile step then using the calling ID would be nice, but `"tmpfs:/pip-cache:rw"` seems to needs root access from my investigations today, and docker-compose doesn't yet support `"tmpfs:/pip-cache:rw,uid=${myid}"` which is a shame. But maybe you see an alternative? I did spend some significant time on this thinking it was possible but I didn't get there. So perhaps what we've got is good enough for now? I feel it one-step further on than previous.

Reply:
Works for me!

Reply:
> But maybe you see an alternative?

Actually this works for me:

```
docker-compose run --workdir /src/requirements \
    -e XDG_CACHE_HOME=/tmp/pip-cache \
    --rm  \
    --no-deps \
    --user=$(CALLER_UID) \
    --entrypoint bash archivematica-storage-service \
        -c "make clean && make all"
```

I guess that `tmpfs` makes things faster but it doesn't seem to be significant.

Reply:
@sevein 7ad4017 works. On the face of it, quite superior in how much code it removes - but I think we may be approaching this with different intentions. 

When you say works, are the results the same? For example, once I've used the updated recipe, my `.txt` permissions are `${USER}:root`.

My approach to this PR has been to minimize how invasive it is - granted to differing quality? success?  

So:

Revision 1: Uses system level calls to reset ownership to `${USER}:${USER}`. (Captures all users running docker in the docker group, or all users running docker as sudo) but it doesn't quite feel right to me either. 

Revision 2: The compromise is that, users must run docker with their user in the group, but the ownership is reset to what the user expects. _(there's no explaining to do to anyone / i leave things how I found them)._ i.e. `${USER}:${USER}`. 

Revision 3: 7ad4017 is fine, but as mentioned, it does seem to leave that slightly niggly group ownership behind, which doesn't sit right personally, but that's because I know we have a simple command to put  things back to how we found them. 

Given that, at the very least, I'd like to keep the reset recipes. But if we do that, then we might as well keep the `tmpfs` too which does make things slightly quicker like you say. 

Which would leave me to change GID back to UID - that was me misunderstanding some terminology. 

Does sticking with revision 2 work with the variable change, or do you need 7ad4017 or another variant of that which can reset group? I can't see a way to reset group but maybe you have another idea still given the problem statement?

Reply:
> When you say works, are the results the same? For example, once I've used the updated recipe, my .txt permissions are ${USER}:root.

Do you think `${USER}:root` is a consequence of your `sudo docker` workflow? It doesn't happen to me. But I can see how you'd want the reset recipes, that's fine.

> But if we do that, then we might as well keep the tmpfs too which does make things slightly quicker like you say.

Cool, that makes sense.

Reply:
Oh I see what's going on with `${USER}:root`. When we use `docker-compose --user=1000` we're only telling Docker the uid we want to use but the gid is set to zero automatically. If we did `--user=$(CALLER_UID):$(CALLER_GID)` then it would be set properly.

```
$ docker-compose run --no-deps --entrypoint=id --rm --user=1000 archivematica-storage-service -g
0
$ docker-compose run --no-deps --entrypoint=id --rm --user=1000:1000 archivematica-storage-service -g
1000
```

Reply:
>  If we did --user=$(CALLER_UID):$(CALLER_GID) then it would be set properly.
That sounds good, Let me give it a whirl and get the PR updated.